### PR TITLE
KFSPTS-11405 refactor removal of easymock and replace with Mockito

### PR DESCRIPTION
**Pull request by jhulslander**
This is ready for review, but there are two functions in CreateAccountingDocumentServiceImplTest that I had trouble refactoring, buildMockWebTarget and buildMockInvocationBuilder .... I could use some helping mocking these.

A couple of things: 
1) I created a new user story to refactor MockDocumentUtils
2) I updated the pom file to use Powermockito to 1.7.3, which is the latest version used at kualico's current master of Financials: https://github.com/KualiCo/financials/blob/master/pom.xml .  So I think that should be fine.

### CODE REVIEW

**Comment by jhulslander** [comment]
retest this

File: src/test/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocumentTest.java [hunk 1]
```diff
@@ -593,24 +581,21 @@ public void setupMockObjects() {
...
-                    .andStubReturn(Boolean.TRUE);
-            EasyMock.expect(authorizer.canEdit(EasyMock.isA(CuDisbursementVoucherDocument.class), EasyMock.not(EasyMock.eq(authorizedPerson))))
-                    .andStubReturn(Boolean.FALSE);
-            EasyMock.expect(authorizer.getEditModes(EasyMock.isA(CuDisbursementVoucherDocument.class), EasyMock.eq(authorizedPerson),
-                    EasyMock.eq(presentationEditModes))).andStubReturn(authorizationEditModes);
-            EasyMock.replay(authorizer);
-            return authorizer;
+            TransactionalDocumentAuthorizer documentAuthorizer = Mockito.mock(TransactionalDocumentAuthorizer.class);
+            Mockito.when(documentAuthorizer.canEdit(Mockito.any(), Mockito.any())).then(this::determineEditCall);
+            Mockito.when(documentAuthorizer.getEditModes(Mockito.any(), Mockito.any(),Mockito.any())).thenReturn(authorizationEditModes);
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocumentTest.java]: The old mock of this method was checking for a much stricter set of arguments. Should the new code do so?

File: src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java [hunk 2]
```diff
@@ -62,28 +60,26 @@
...
-        accountsPayableService = EasyMock.createNiceMock(AccountsPayableService.class);
-        EasyMock.expect(accountsPayableService.getExpiredOrClosedAccountList(creditMemoDocument)).andReturn(new HashMap<>());
-        EasyMock.replay(accountsPayableService);
-
-        dataDictionaryService = EasyMock.createNiceMock(DataDictionaryService.class);
-        EasyMock.expect(dataDictionaryService.getAttributeMaxLength(DocumentHeader.class, KRADPropertyConstants.DOCUMENT_DESCRIPTION)).andReturn(200);
-        EasyMock.replay(dataDictionaryService);
-
+        accountsPayableService = Mockito.mock(AccountsPayableService.class);
+        Mockito.when(accountsPayableService.getExpiredOrClosedAccountList(creditMemoDocument)).thenReturn(new HashMap<>());
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java]: It looks like creditMemoDocument is null at this point, and I think the old test was the same way. Should this be modified to just check for any argument? Or could this be placed after the document has been set up?

File: src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java [hunk 3]
```diff
@@ -62,28 +60,26 @@
...
-        purapService = EasyMock.createMock(PurapService.class);
+        documentService = buildMockDocumentService();
+        noteService = Mockito.mock(NoteService.class);
+        purapService = Mockito.mock(PurapService.class);
         vendorService = new MockVendorServiceImpl();
 
-        creditMemoServiceImpl = PowerMock.createPartialMock(CuCreditMemoServiceImplTest.TestCuCreditMemoServiceImpl.class, "reIndexDocument", "getCreditMemoDocumentById");
+        
+        creditMemoServiceImpl = PowerMockito.spy(new TestCuCreditMemoServiceImpl());
+        PowerMockito.doNothing().when(creditMemoServiceImpl, "reIndexDocument", Mockito.any());
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java]: I see that "getCreditMemoDocumentById" is no longer being mocked. Did it not need mocking anymore (possibly due to "nice" mocking functionality in Mockito)?

File: src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java [hunk 4]
```diff
@@ -99,16 +95,28 @@ public void setUp() throws Exception {
...
+    }
+    private Note buildDocumentNote(InvocationOnMock invocation) {
+        Document document = invocation.getArgument(0);
+        String noteText = invocation.getArgument(1);
+        
+        Note note = Mockito.mock(Note.class);
+        Mockito.when(note.getNotePostedTimestamp()).thenReturn(dateTimeService.getCurrentTimestamp());
+        Mockito.when(note.getVersionNumber()).thenReturn(Long.valueOf(1L));
+        Mockito.when(note.getNoteText()).thenReturn(noteText);
+        Mockito.when(note.getNoteTypeCode()).thenReturn("TST");
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java]: Is this the correct note type to use? Or does this test not validate the note type very strictly?

File: src/test/java/edu/cornell/kfs/sys/util/MockPersonUtil.java [hunk 5]
```diff
@@ -1,29 +1,35 @@
...
-        UserSession userSession = EasyMock.createMock(UserSession.class);
-        EasyMock.expect(userSession.getPrincipalId()).andStubReturn(person.getPrincipalId());
-        EasyMock.expect(userSession.getPrincipalName()).andStubReturn(person.getPrincipalName());
-        EasyMock.expect(userSession.getLoggedInUserPrincipalName()).andStubReturn(person.getPrincipalName());
-        EasyMock.expect(userSession.getPerson()).andStubReturn(person);
-        EasyMock.expect(userSession.getActualPerson()).andStubReturn(person);
-        EasyMock.replay(userSession);
+        UserSession userSession = mock(UserSession.class);
+        
+        String principleId = person.getPrincipalId();
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/sys/util/MockPersonUtil.java]: Typo in variable name.

File: src/test/java/edu/cornell/kfs/sys/util/MockPersonUtil.java [hunk 6]
```diff
@@ -1,29 +1,35 @@
...
+        when(userSession.getPrincipalId()).thenReturn(principleId);
+        
+        String principleName = person.getPrincipalName();
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/sys/util/MockPersonUtil.java]: Typo in variable name.

File: src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java (hunk 2 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java]: I changed it to Mockito.any

File: src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java (hunk 4 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java]: Note type isn't validated strictly

File: src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java (hunk 3 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java]: I was looking back at this, it looks like getCreditMemoDocumentById doesn't need to be mocked any longer to meet the needs of this test.

**Comment by jhulslander** [comment]
I was able to refactor CreateAccountingDocumentServiceImplTest by using method references on the build mock functions, and then use straight Mockito, and not powermockito

This is ready for review again

File: src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java (hunk 3 above)
Comment (cah292) [reply src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java]: Ok, thanks for checking.

File: src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java (hunk 4 above)
Comment (cah292) [reply src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java]: Ok, thanks for confirming.

**Pull request merged**